dot/network/state.go

```diff
@@ -34,6 +34,8 @@ type Syncer interface {
 
 	// CreateBlockResponse is called upon receipt of a BlockRequestMessage to create the response
 	CreateBlockResponse(peer.ID, *messages.BlockRequestMessage) (*messages.BlockResponseMessage, error)
+
+	OnConnectionClosed(peer.ID)
```

Comment:
Can you add a comment here about why we need this?

Reply:
Comment added: 

```
// OnConnectionClosed should be trigged whenever Gossamer closes a connection with another
// peer, normally used when the peer reputation is too low.
```